File: homeworks/lesson3/Figures/Square.java
package homeworks.lesson3.Figures;

import homeworks.lesson3.Exceptions.IncorrectTriangleException;
import homeworks.lesson3.Exceptions.NegativeValueException;

public class Square extends Rectangle{
    
    public Square(String side) throws NegativeValueException, IncorrectTriangleException{
        super(side, side);
    }

    public Square() throws NegativeValueException, IncorrectTriangleException{
        this("6");
    }

    @Override
    public String toString() {
        return String.format("Квадрат\nСторона: %s", sides[0]);
    }
}

File: homeworks/lesson6/Figures/Square.cs
public class Square : Rectangle
{
    public Square(int a) : base(a, a)
    {

    }

    public Square() : base(5, 5)
    {

    }

    public override string ToString()
    {
        return "Квадрат\nСторона: " + base.sides[0];
    }
}
